services/brig/src/Brig/Data/Nonce.hs

```diff
@@ -0,0 +1,52 @@
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2022 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module Brig.Data.Nonce
+  ( addNonce,
+    invalidateNonce,
+  )
+where
+
+import Brig.App (Env)
+import Brig.Data.Instances ()
+import Cassandra
+import Control.Lens hiding (from)
+import Data.Nonce (Nonce)
+import Imports
+
+addNonce ::
+  (MonadClient m, MonadReader Brig.App.Env m) =>
+  Int ->
+  Nonce ->
+  m ()
+addNonce ttl nonce = retry x5 . write insertNonce $ params LocalQuorum (Identity nonce)
+  where
+    insertNonce :: PrepQuery W (Identity Nonce) ()
+    insertNonce = fromString $ "INSERT INTO nonce (nonce) VALUES (?)" <> renderTtl
+    renderTtl :: String
+    renderTtl
+      | ttl > 0 = " USING TTL " <> show ttl
+      | otherwise = " USING TTL null"
```

Comment:
I'm not sure `null` works, but I'm sure `0` works.  So the only thing you want to handle here are negative numbers, and you can avoid those by using `Word64`, which fits 6 years in seconds.  Then you can use a simple `show` or `toByteString'` instead of this render function.